haystack/nodes/image_to_text/base.py

```diff
@@ -0,0 +1,55 @@
+from typing import List, Optional
+
+from abc import abstractmethod
+
+from haystack.schema import Document
+from haystack.nodes.base import BaseComponent
+
+
+class BaseImageToText(BaseComponent):
+    """
+    Abstract class for ImageToText
+    """
+
+    outgoing_edges = 1
+
+    @abstractmethod
+    def generate_captions(
+        self, image_file_paths: List[str], generate_kwargs: Optional[dict] = None, batch_size: Optional[int] = None
+    ) -> List[Document]:
+        """
+        Abstract method for generating captions.
+
+        :param image_file_paths: Paths of the images
+        :param generate_kwargs: Dictionary containing arguments for the generate method of the Hugging Face model.
```

Comment:
I see why you named `generate_kwargs` like this, and I'm up for discussing it, but I'd rather call this parameter `generation_kwargs`. At a first glance, `generate_kwargs` sounded like it was a boolean parameter instructing the node to generate some kwargs :smile: